src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java

```diff
@@ -0,0 +1,115 @@
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.intellij.quarkus.telemetry;
+
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.application.ApplicationManager;
+import com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder;
+import com.redhat.devtools.intellij.telemetry.core.util.Lazy;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Helper to initialize Telemetry. A Telemetry service based on <a href="https://github.com/redhat-developer/intellij-redhat-telemetry">intellij-redhat-telemetry</a>,
+ * will be instantiated if that plugin is installed, a No-Op Telemetry stub will be used otherwise.
+ */
+public class TelemetryManager implements Disposable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TelemetryManager.class);
+
+    private final Lazy<TelemetryMessageBuilder> builder = new Lazy<>(() -> new TelemetryMessageBuilder(PluginManager.getPluginByClass(this.getClass())));
+
+    private boolean hasError;
+
+    public static TelemetryManager instance() {
+        return ApplicationManager.getApplication().getService(TelemetryManager.class);
+    }
+
+    /**
+     * Sends a tracking event without additional properties.
+     * @param eventName the name of the event
+     */
+    public void send(@NotNull TelemetryEventName eventName) {
+        send(eventName, null);
+    }
+
+    /**
+     * Sends a tracking event with additional properties.
+     * @param eventName the name of the event
+     * @param properties the properties of the event
+     */
+    public void send(@NotNull TelemetryEventName eventName, @Nullable Map<String, String> properties) {
+        TelemetryMessageBuilder.ActionMessage action = action(eventName, properties);
+        if (action == null) {
+            return;
+        }
+        asyncSend(action);
+    }
+
+    @Nullable
+    public TelemetryMessageBuilder.ActionMessage action(@NotNull TelemetryEventName eventName) {
+        return action(eventName, null);
+    }
+
+    @Nullable
+    public TelemetryMessageBuilder.ActionMessage action(@NotNull TelemetryEventName eventName, @Nullable Map<String, String> properties) {
+        TelemetryMessageBuilder builder = getMessageBuilder();
+        if (builder == null) {
+            return null;
+        }
+        TelemetryMessageBuilder.ActionMessage action = builder.action(eventName.getEventName());
```

Comment:
if builder.action API changed, that'd crash here, error wouldn't be caught

Reply:
fixed